tests/Test.ADAL.NET.Integration/AcquireTokenSilentTests.cs

```diff
@@ -0,0 +1,113 @@
+
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Test.ADAL.Common;
+using Test.ADAL.NET.Unit.Mocks;
+
+namespace Test.ADAL.NET.Unit
+{
+    [TestClass]
+    public class AcquireTokenSilentTests
+    {
+        [TestInitialize]
+        public void Initialize()
+        {
+            HttpMessageHandlerFactory.ClearMockHandlers();
+            InstanceDiscovery.InstanceCache.Clear();
+            HttpMessageHandlerFactory.AddMockHandler(MockHelpers.CreateInstanceDiscoveryMockHandler());
+        }
+
+        [TestMethod]
+        [TestCategory("AcquireTokenSilentTests")]
+        public void AcquireTokenSilentServiceErrorTestAsync()
+        {
+            TokenCache cache = new TokenCache();
+            TokenCacheKey key = new TokenCacheKey(TestConstants.DefaultAuthorityCommonTenant,
+                TestConstants.DefaultResource, TestConstants.DefaultClientId, TokenSubjectType.User, "unique_id",
+                "displayable@id.com");
+            cache.tokenCacheDictionary[key] = new AuthenticationResultEx
+            {
+                RefreshToken = "something-invalid",
+                ResourceInResponse = TestConstants.DefaultResource,
+                Result = new AuthenticationResult("Bearer", "some-access-token", DateTimeOffset.UtcNow)
+            };
+
+            AuthenticationContext context = new AuthenticationContext(TestConstants.DefaultAuthorityCommonTenant, cache);
+
+            var ex = AssertException.TaskThrows<AdalSilentTokenAcquisitionException>(async () =>
+                {
+                    HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler()
+                    {
+                        Method = HttpMethod.Post,
+                        ResponseMessage = MockHelpers.CreateInvalidGrantTokenResponseMessage()
+                    });
+                    await context.AcquireTokenSilentAsync(TestConstants.DefaultResource, TestConstants.DefaultClientId, new UserIdentifier("unique_id", UserIdentifierType.UniqueId));
+                });
+            
+            Assert.AreEqual(AdalError.FailedToAcquireTokenSilently, ex.ErrorCode);
+            Assert.AreEqual(AdalErrorMessage.FailedToRefreshToken, ex.Message);
+            Assert.IsNotNull(ex.InnerException);
+            Assert.IsTrue(ex.InnerException is AdalException);
+            Assert.AreEqual(((AdalException)ex.InnerException).ErrorCode, "invalid_grant");
+
+        }
+
+        [TestMethod]
+        [TestCategory("AcquireTokenSilentTests")]
+        //292916 Ensure AcquireTokenSilent tests exist in ADAL.NET for public clients
+        public async Task AcquireTokenSilentTestWithValidTokenInCache()
+        {
+            var context = new AuthenticationContext(TestConstants.DefaultAuthorityHomeTenant, true);
+            context.TokenCache.Clear();
```

Comment:
You can choose to pass in a cache instance via constructor instead. That will make the test resilient to such issues in the future.